runelite-client/src/main/java/net/runelite/client/plugins/sticktothescript/gecooker/GECookerScript.java

```diff
@@ -0,0 +1,261 @@
+            GEWorkLocation.SOUTH_EAST.getWorldPoint(),
+            GEWorkLocation.NORTH_WEST.getWorldPoint(),
+            GEWorkLocation.SOUTH_WEST.getWorldPoint()
+    );
+    private static List<Integer> FireIDs = Arrays.asList(26185, 49927);
+
+    public boolean run(GECookerConfig config) {
+        CookingItem cookingItem = config.sCookItem();
+        LogType logType = config.sLogType();
+        GEWorkLocation desiredLocation = config.sLocation();
+
+        Microbot.enableAutoRunOn = false;
+        mainScheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            if (!super.run() || !Microbot.isLoggedIn()) {
+                debug("Not running");
+                return;
+            }
+
+            if (Rs2AntibanSettings.actionCooldownActive) {
+                debug("Cool down active");
+                return;
+            }
+
+            determineState(cookingItem, logType);
+
+            if (Rs2Dialogue.hasContinue()) {
+                debug("Click to continue");
+                Rs2Dialogue.clickContinue();
+                return;
+            }
+
+           switch (state) {
+            case COOKING:
+                if (Rs2Inventory.count(cookingItem.getRawItemID()) == 0) {
+                    debug("Out of cooking item in inventory");
+                    return;
+                }
+
+                if (Rs2Player.waitForXpDrop(Skill.COOKING, 3500)) {
+                    debug("Interacting");
+                    Rs2Antiban.actionCooldown();
+                    Rs2Antiban.takeMicroBreakByChance();
+                    return;
+                }
+
+                WorldPoint activeFireLocation = Firemaking.isGameObjectOnTile(FireSpots, FireIDs);
+
+                if (activeFireLocation != null && Rs2Player.distanceTo(activeFireLocation) > 3) {
+                    debug("Walking to existing fire");
+                    Rs2Walker.walkTo(activeFireLocation);
+                    sleep(180, 540);
+                    return;
+                }
+
+                debug("Looking for fires to use");
+
+                boolean interacted = false;
+                for (WorldPoint FireSpot : FireSpots) {
+                    TileObject fireTile = Rs2GameObject.findGameObjectByLocation(FireSpot);
+                    if (fireTile != null && FireIDs.contains(fireTile.getId())) {
+                        debug("Using object on fire");
+                        Rs2Inventory.useItemOnObject(cookingItem.getRawItemID(), fireTile.getId());
+                        interacted = true;
+                        break;
+                    }
+                }
+
+                if (interacted) {
+                    sleepUntil(() -> !Rs2Player.isMoving() && Rs2Widget.findWidget("How many would you like to cook?", null, false) != null, 5000);
+                    sleep(180, 540);
+                    Rs2Keyboard.keyPress(KeyEvent.VK_SPACE);
+                    Rs2Player.waitForXpDrop(Skill.COOKING, 5000);
```

Comment:
With this logic, the loop would reset once one item was cooked? Cooking XP is given after each item is cooked

Reply:
That is correct. There is logic at the top of the `COOKING` case that handles the looping logic. This way, the logic can be simplified and not stuck in an infinite loop.

Reply:
Yes, I see your determineState method, but wouldn't want to wait until the entire inventory has finished processing until you reset the loop? otherwise this would use each individual raw item on the fire resulting in a lot of clicking

Reply:
Nope! That's the purpose of the section:

```java
if (Rs2Inventory.count(cookingItem.getRawItemID()) == 0) {
    debug("Out of cooking item in inventory");
    return;
}

if (Rs2Player.waitForXpDrop(Skill.COOKING, 3500)) {
    debug("Interacting");
    Rs2Antiban.actionCooldown();
    Rs2Antiban.takeMicroBreakByChance();
    return;
}
```

The first check is redundant because the determineState function should catch it, but I left it in as a failsafe. The second check waits for the XP drop for 3.5 seconds. Since this function is blocking, during each iteration, we are essentially waiting for an XP drop. If we go longer than the 3.5 seconds, we can assume that we are not cooking. From my testing, the time it took to between cooking two pieces of food was around 2.5 - 3 seconds.

Reply:
I apologize if my logic seems a little strange. I primarily work with real-time operating systems. Typically with RTOS, you want to avoid getting caught in too many loops and get back to the main loop and reiterate as many times as possible.

Reply:
No this completely make sense, you're utilizing the outer loop by removing any prerequisites conditions by forcing it to go somewhere else. 

I was more of curious was all!